File: src/main/java/com/nedap/university/go/gocommands/clientToServer/ClientChatCommand.java
package com.nedap.university.go.gocommands.clientToServer;

import com.nedap.university.go.gocommands.Command;
import com.nedap.university.go.server.ClientHandler;

/**
 * Created by martijn.slot on 21/02/2017.
 */
public class ClientChatCommand extends Command {

    private ClientHandler clientHandler;

    public ClientChatCommand(String[] command, ClientHandler clientHandler) {
        super();
        this.splitMessage = command;
        this.clientHandler = clientHandler;
    }

    @Override
    public void execute() {
        switch (clientHandler.getClientStatus()) {
            case PREGAME:
            case WAITING:
                clientHandler.chatToAll(splitMessage);
            case INGAME_NOT_TURN:
            case INGAME_TURN:
                clientHandler.chatToOpponent(splitMessage);
                break;
            default:
                cannotExecute();
                break;
        }

    }

    protected void cannotExecute() {
        clientHandler.writeToClient("WARNING Cannot CHAT, you've broken the program if you've reached this position. ");
    }

}

File: src/main/java/com/nedap/university/go/gocommands/clientToServer/ClientChatCommand.kt
package com.nedap.university.go.gocommands.clientToServer

import com.nedap.university.go.gocommands.Command
import com.nedap.university.go.server.ClientHandler

/**
 * Created by martijn.slot on 21/02/2017.
 */
class ClientChatCommand(command: Array<String>, private val clientHandler: ClientHandler) : Command() {

    init {
        this.splitMessage = command
    }

    @Override
    fun execute() {
        when (clientHandler.getClientStatus()) {
            PREGAME, WAITING -> clientHandler.chatToAll(splitMessage)
            INGAME_NOT_TURN, INGAME_TURN -> clientHandler.chatToOpponent(splitMessage)
            else -> cannotExecute()
        }

    }

    protected fun cannotExecute() {
        clientHandler.writeToClient("WARNING Cannot CHAT, you've broken the program if you've reached this position. ")
    }

}
